### PR TITLE
Use `UserContextApi` for `UserContext` access

### PR DESCRIPTION
**Pull request by YanWQ-monad**
As title. Make `aster-nix` modify `UserContext` registers through `UserContextApi`, instead of accessing the registers directly.

This could help to reduce the degree of architecture-related coupling.

### CODE REVIEW

**Review by junyang-zh — COMMENTED** [review]
This PR is in a good shape. I just feel a little bit awkward for the naming of the `UserContextApi` trait, which is an issue of the legacy code. By browsing [discussions of trait naming convensions](https://github.com/rust-lang/api-guidelines/discussions/28), I think it should be either a noun or a verb. Maybe `UserContextAccess`.

**Comment by junyang-zh** [comment]
`UserContextAbi` may be another good alternative. Since "which reg is used for arguments" is defined by the ABI.

**Comment by YanWQ-monad** [comment]
I think `UserContextAbi` make sense, at least I have no idea about a better name...

**Review by junyang-zh — APPROVED** [review]
This PR looks good to me now, as further changes to support other ABI standards is not planned and out of scope of this PR. Need @tatetian for further comments.

File: kernel/aster-nix/src/arch/mod.rs [hunk 1]
```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: MPL-2.0
+
+#[cfg(target_arch = "x86_64")]
+mod x86;
+
+pub trait LinuxAbi {
```
Comment (tatetian) [inline kernel/aster-nix/src/arch/mod.rs:6]: The context of `LinuxAbi` is the `arch` module. From this context, it is unclear that `LinuxAbi` refers to the CPU-related part of the Linux ABI. The notion of the ABI of a kernel covers other aspects of the kernel in addition to the CPU. So it needs to have a more accurate name (e.g., `LinuxCpuContext`) or a more accurate context (e.g., adding a `cpu` module).

File: kernel/aster-nix/src/arch/mod.rs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    fn syscall_ret(&self) -> usize;
+
+    /// Set return value of syscall
+    fn set_syscall_ret(&mut self, ret: usize);
+
+    /// Get syscall args
+    fn syscall_args(&self) -> [usize; 6];
+
+    /// Set function argument
+    fn set_argument(&mut self, index: usize, value: usize);
```
Comment (tatetian) [inline kernel/aster-nix/src/arch/mod.rs:20]: Looking at this trait along, it is unclear what arguments this method set. And it looks like this method is only useful for setting the arguments of a signal handler and no other parts of the kernel are interested about it. So I suggest moving this knowledge inside `signal/mod.rs`. Hiding such knowledge from the rest of the system makes the code easier to understand and maintain.

---

Edit: consider adding two modules under `arch`: one is `cpu.rs`, the other is `signal.rs`. And then add appropriate arch-independent abstractions or arch-dependent implementations to them.

**Review by tatetian — APPROVED** [review]
LGTM. Thanks for the contribution!

**Comment by tatetian** [comment]
This error message is repeated for countless times in system call test.

> Error: : page fault handler failed: addr: 0x0, err: Error { errno: EACCES, msg: Some("page fault addr is not in current vmar") }

I suspect that this PR introduces a bug. Could you examine the modifications?

**Comment by tatetian** [comment]
> But, it seems beyond the scope of this PR? and may need further effort.

I can't see the last version of the PR has caused the problem. I am counting on you to figure out the root cause ❤️

**Pull request merged**